csharp/private/rules/resgen.bzl

```diff
@@ -0,0 +1,50 @@
+# Label of the template file to use.
+_TEMPLATE = "@d2l_rules_csharp//csharp/private:rules/Template.csproj"
+_TEMPLATE_EMBEDDED_RESOURCE = "        <EmbeddedResource Include=\"%s\" />"
+
+def _csproj_embedded_resource(resx_files):
+    result = ""
+    for src in resx_files:
+        result += _TEMPLATE_EMBEDDED_RESOURCE % (src.basename)
+    return result
+
+def _csharp_resx_impl(ctx):
+    proj_name = ctx.file.csproj.basename[:-(len(ctx.file.csproj.extension)+1)]
+
+    csproj_output = ctx.actions.declare_file(ctx.file.csproj.basename)
+    embedded_resources = _csproj_embedded_resource(ctx.files.srcs)
+    ctx.actions.expand_template(
+        template = ctx.file._template,
+        output = csproj_output,
+        substitutions = {
+            "{FRAMEWORK}": ctx.attr.target_frameworks[0],
+            "{RESOURCES}": embedded_resources,
+        },
+    )
+
+    files = depset(direct = csproj_output)
+    runfiles = ctx.runfiles(files = csproj_output)
+    return [DefaultInfo(files = files, runfiles = runfiles)]
+
+csharp_resx = rule(
+    implementation = _csharp_resx_impl,
+    attrs = {
+        "srcs": attr.label_list(
+            mandatory = True, 
+            allow_files = True
+        ),
+        "csproj": attr.label(

```

Comment:
I am adding this attribute as a placeholder (it allows it to work in my WIP branch) for what I believe might need to be namespacing information. When using the name of the rule for the `.csproj` (hello.strings.csproj), it resulted in the compiled CLR binary not being discovered when compiling the `csharp_binary`. 

I suspect the reason for that is that the namespace is not set in the `csproj` template. It then must use the name of the  csproj as its namespace. I will be looking into this later when I get a chance.

Reply:
I think I can avoid this issue if I make use of [identifier](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/resource-compiler-option) when passing the resource file to msbuild